core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -324,14 +337,62 @@ public JDBCUserStoreManager(RealmConfiguration realmConfig, Map<String, Object>
         }
 
         initUserRolesCache();
-
+        if (hashProvider == null) {
+            hashProvider = initializeHashProvider(realmConfig.getUserStoreProperties());
+        }
         if (log.isDebugEnabled()) {
             log.debug("Ended " + System.currentTimeMillis());
         }
         /* Initialize user roles cache as implemented in AbstractUserStoreManager */
 
     }
 
+    /**
+     * Initializes the HashProvider.
+     *
+     * @param userStorePropertiesMap The map which contains the params needed to be initialized and the digest function.
+     * @return Initialized HashProvider.
+     */
+    private HashProvider initializeHashProvider(Map<String, String> userStorePropertiesMap) {
+
+        String digestFunction = userStorePropertiesMap.get(JDBCRealmConstants.DIGEST_FUNCTION);
+        HashProviderFactory hashProviderFactory =
+                UserStoreMgtDataHolder.getInstance().getHashProviderFactory(digestFunction);
+        if (hashProviderFactory == null) {
+            hashProvider = null;
+        } else {
+            List<String> metaProperties = hashProviderFactory.getHashProviderMetaProperties();
+            Map<String, Object> hashProviderPropertiesMap = new HashMap<>();
+            if (!metaProperties.isEmpty()) {
+                hashProviderPropertiesMap = getHashProviderInitConfigs(userStorePropertiesMap, metaProperties);
+            }
+            hashProvider = hashProviderFactory.getHashProvider(hashProviderPropertiesMap);
```

Comment:
If the `metaProperties ` is empty can't we use the method with no arguments?

Reply:
resolved